service/src/main/java/bio/terra/workspace/service/resource/controlled/flight/clone/workspace/CloneCreateCloudContextFlight.java

```diff
@@ -0,0 +1,107 @@
+    var userRequest =
+        FlightUtils.getRequired(
+            inputParameters,
+            JobMapKeys.AUTH_USER_INFO.getKeyName(),
+            AuthenticatedUserRequest.class);
+    var sourceWorkspaceId =
+        FlightUtils.getRequired(
+            inputParameters, ControlledResourceKeys.SOURCE_WORKSPACE_ID, UUID.class);
+    var spendProfile =
+        inputParameters.get(WorkspaceFlightMapKeys.SPEND_PROFILE, SpendProfile.class);
+
+    addStep(new CreateCloudContextIdsForFutureStepsStep());
+
+    // Only create a cloud context if the source workspace has a cloud context
+    Optional<GcpCloudContext> gcpCloudContextOptional =
+        flightBeanBag.getGcpCloudContextService().getGcpCloudContext(sourceWorkspaceId);
+    if (gcpCloudContextOptional.isPresent()) {
+      SpendProfile gcpSpendProfile =
+          getSpendProfile(spendProfile, gcpCloudContextOptional.get(), flightBeanBag, userRequest);
+
+      addStep(
+          new LaunchCreateCloudContextFlightStep(
+              flightBeanBag.getWorkspaceService(),
+              CloudPlatform.GCP,
+              gcpSpendProfile,
+              ControlledResourceKeys.CREATE_GCP_CLOUD_CONTEXT_FLIGHT_ID),
+          cloudRetryRule);
+      addStep(
+          new AwaitCreateCloudContextFlightStep(
+              ControlledResourceKeys.CREATE_GCP_CLOUD_CONTEXT_FLIGHT_ID),
+          longCloudRetryRule);
+    }
+
+    Optional<AzureCloudContext> azureCloudContextOptional =
+        flightBeanBag.getAzureCloudContextService().getAzureCloudContext(sourceWorkspaceId);
+    if (azureCloudContextOptional.isPresent()) {
+      SpendProfile azureSpendProfile =
+          getSpendProfile(
+              spendProfile, azureCloudContextOptional.get(), flightBeanBag, userRequest);
+      addStep(
+          new LaunchCreateCloudContextFlightStep(
+              flightBeanBag.getWorkspaceService(),
+              CloudPlatform.AZURE,
+              azureSpendProfile,
+              ControlledResourceKeys.CREATE_AZURE_CLOUD_CONTEXT_FLIGHT_ID),
+          cloudRetryRule);
+      addStep(
+          new AwaitCreateCloudContextFlightStep(
+              ControlledResourceKeys.CREATE_AZURE_CLOUD_CONTEXT_FLIGHT_ID),
+          cloudRetryRule);
+    }
+  }
+
+  /*
+   * There is an API mismatch in the V1 APIs. If a new spend profile is passed with
+   * the clone workspace, then we use it to create all the new cloud contexts,
+   * even though a spend profile is only valid for one cloud. Since we don't have
+   * multi-cloud workspaces right now, we will not have a mismatch of spend profile
+   * and cloud context.
+   */
+  private SpendProfile getSpendProfile(
+      SpendProfile spendProfile,
+      CloudContext cloudContext,
+      FlightBeanBag flightBeanBag,
+      AuthenticatedUserRequest userRequest) {
+    if (spendProfile != null) {
+      return spendProfile;
+    }
+
+    var spendProfileService = flightBeanBag.getSpendProfileService();
```

Comment:
@marctalbott It looks like all this code is no longer necessary because you introduced it further upstream, [here](https://github.com/DataBiosphere/terra-workspace-manager/blob/cfdaa8e366001b33b2cfa0a8c63710f1fd8e37af/service/src/main/java/bio/terra/workspace/app/controller/WorkspaceApiController.java#L623).